demonstrations/tutorial_learningshallow.py

```diff
@@ -0,0 +1,433 @@
+r"""Learning shallow quantum circuits with local inversions and circuit sewing
+==============================================================================
+
+In a recent paper `Learning shallow quantum circuits <https://arxiv.org/abs/2401.10095>`_ [#Huang]_, Huang et al
+introduce and prove performance bounds on efficient algorithms to learn constant depth circuits.
```

Comment:
```suggestion
prove performance bounds on efficient algorithms to reconstruct constant-depth circuits.
```
* shorter
* explains what learning means by using an alternative word to their title
* consider hyphen everywhere, I'm not sure though